proof-systems/src/darlin/accumulators/dlog.rs

```diff
@@ -0,0 +1,743 @@
+        // We do final verification and the batching of the GFin in a single MSM
+        let hard_time = start_timer!(|| "Batch verify hard parts");
+        let final_val = InnerProductArgPC::<G, D>::cm_commit(
+            // The vk might be oversized, but the VariableBaseMSM function, will "trim"
+            // the bases in order to be as big as the scalars vector, so no need to explicitly
+            // trim the vk here.
+            &[final_comm_keys.as_slice(), vk.comm_key.as_slice()].concat(),
+            &[batching_chal_pows.as_slice(), combined_check_poly.coeffs.as_slice()].concat(),
+            None,
+            None,
+        );
+        end_timer!(hard_time);
+
+        if !ProjectiveCurve::is_zero(&final_val) {
+            end_timer!(check_time);
+            return Ok(false);
+        }
+        end_timer!(check_time);
+        Ok(true)
+    }
+
+    /// Our implementation is size optimized: the accumulation proof is all that it's needed since
+    /// that the g_fin of the new accumulator can be derived from the AccumulationProof (which is
+    /// a DLOG opening proof) and the xi_s can be recomputed via succinct verification of the previous
+    /// accumulators.
+    fn accumulate_items(
+        ck: &Self::AccumulatorProverKey,
+        accumulators: Vec<Self::Item>,
+    ) -> Result<(Self::Item, Self::AccumulationProof), Error>
+    {
+        let accumulate_time = start_timer!(|| "Accumulate");
+
+        // Sample a new challenge point z
+        let z = InnerProductArgPC::<G, D>::compute_random_oracle_challenge(
+            &to_bytes![ck.hash.clone(), accumulators.as_slice()].unwrap(),
+        );
+
+        // Collect GFinals from the accumulators
+        let g_fins = accumulators.iter().map(|acc| {
+            Commitment::<G> {
+                comm: acc.g_final.comm.clone(),
+                shifted_comm: None
+            }
+        }).collect::<Vec<_>>();
+
+        // Collect xi_s from the accumulators
+        let xi_s = accumulators.into_iter().map(|acc| {
+            acc.xi_s
+        }).collect::<Vec<_>>();
+
+        let poly_time = start_timer!(|| "Open Bullet Polys");
+
+        // Compute multi poly single point opening proof of the Bullet polys
+        // at the GFin(s)
+        let opening_proof = InnerProductArgPC::<G, D>::open_check_polys(
+            &ck,
+            xi_s.iter(),
+            g_fins.iter(),
+            z
+        )?;
+
+        end_timer!(poly_time);
+
+        // Even if our implementation is size optimized, the API requires us to
+        // return an accumulator too: so we return a dummy one instead (to be
+        // discarded by the caller).
+        let accumulator = DLogItem::<G>::default();
+
+        let mut accumulation_proof = AccumulationProof::<G>::default();
+        accumulation_proof.pc_proof = opening_proof;
+
+        end_timer!(accumulate_time);
+
+        Ok((accumulator, accumulation_proof))
+    }
+
+    fn verify_accumulated_items<R: RngCore>(
```

Comment:
We probably had this a couple of times before. Why don't we call it just verify_accumulation()?